pkg/events/derive/derive.go

```diff
@@ -0,0 +1,64 @@
+package derive
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// eventSkeleton is a struct for the necessary information from an event definition to create the derived event
+type eventSkeleton struct {
+	Name   string
+	ID     int
+	Params []trace.ArgMeta
+}
+
+type argsGenerator func(event trace.Event) ([]interface{}, error)
+
+// generateSingleEventDeriveFunction create an events.DeriveFunction to generate a single derive trace.Event.
+// The event will be created using the original event information, the ID given and the arguments given.
+// The order of the arguments given will match the order in the event definition, so make sure the order match
+// the order of the params in the events.event struct of the event under events.Definitions.
+func generateSingleEventDeriveFunction(id events.ID, argsGen argsGenerator) events.DeriveFunction {
```

Comment:
This is a nice abstraction IMO.
But maybe just `singleEventDerive` or `singleEventDeriveFunc`? Generate usually implies to me long term generator functions (the kinds that in python for example would `yield`).